src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -919,9 +938,79 @@ - (void)_handleSubscriptionError:(NSError *)error
     [self _changeState:MTRDeviceStateUnreachable];
 }
 
-- (void)_handleResubscriptionNeeded
+// This method is used for signaling whether to use the subscription pool. This functions as
+// a heuristic for whether to throttle subscriptions to the device via a pool of subscriptions.
+// If products appear that have both Thread and Wifi enabled but is primarily on wifi, this
+// method will need to be updated to reflect that.
+- (BOOL)_deviceUsesThread
 {
-    std::lock_guard lock(_lock);
+    os_unfair_lock_assert_owner(&self->_lock);
+
+    // Device is thread-enabled if there is a Thread Network Diagnostics cluster on endpoint 0
```

Comment:
Thread Network Diagnostics is optional, no?

What we should really be doing is checking for the FeatureMap of the network commissioning cluster on the root endpoint, I would think.

Reply:
Not for thread devices, no.

Reply:
In the spec it is required for Thread devices, same as WiFi Diagnostics for Wi-Fi Devices.

Reply:
Where is it required?  The conformance for it on Root Node device type is `[Thread]`, which means "optionally allowed if Thread".

Reply:
I do agree. Thread Diagnostic cluster is optional, we should not check whether the device operates on Thread based on this cluster.

Reply:
My bad! I read it wrong, Jeff is moving it to another thing.

Reply:
I've changed this logic to look at NetworkCommissioning cluster feature map, and look for the thread bit.